syntax = "proto3";

service ScanService {
  rpc Scan(ScanParameter) returns (stream ScanResult) {}
//  rpc GetStream(ScanParameter) returns (stream ScanResult) {}
}

message ScanParameter {
  // todo
  string scan_mod = 1;
  string scan_host_range = 2;
  string scan_host_port = 3;
  string scan_host_exclude = 4;
  string scan_speed = 5;
  string scan_delay = 6;
  string scan_host_timeout = 7;
  string scan_args = 8;
}

message ScanResult {
  // bool status = 1;
  string ip = 1;
  string scan_result = 2;
}